### ISSUE TITLE
Rule: Use `any` instead of `find`

### ISSUE THREAD

**Issue by BraisGabin** [detekt/detekt#4243]
## Expected Behavior of the rule
Use `any { /* predicate */ }` instead of `find { /* predicate */ } != null`

sample:

```kotlin
// non-compilant
listOf("static", "code").find { it.lenght == 4 } != null
// compilatn
listOf("static", "code").any { it.lenght == 4 }
```

## Context
I was making this "mistake" on #4241 and it should be easily sppoted by detekt.

For sure this is a style rule.

**Comment by TWiStErRob** [comment]
I'm surprised IDEA didn't warn about this, they have a lot of simplification refactor suggestions.

Note mind that there are many extension functions:
Sequence, Iterable, Array, BooleanArray, ByteArray, CharArray, DoubleArray, FloatArray, IntArray, LongArray, ShortArray, Flow?
